app/components/file_upload_preview/file_upload_item/file_upload_item.js

```diff
@@ -134,7 +136,12 @@ export default class FileUploadItem extends PureComponent {
 
         Client4.trackEvent('api', 'api_files_upload');
 
-        this.uploadPromise = RNFetchBlob.fetch('POST', Client4.getFilesRoute(), headers, data);
+        const certificate = await mattermostBucket.getPreference('cert', LocalConfig.AppGroupId);
```

Comment:
Kind of related to my comment on another PR, it would be nice to pull this into redux so that we don't have to turn every function using it into an async function. I'm worried about us overusing async functions since it's easy to mistake them for being synchronous and then getting caught when they don't complete as you expected.

Nothing to do now, but something to think about for later

Reply:
we could although in this case I'm not that sure, but as with the other PR, feel free to create an improvement ticket